core/Services/Calendar/Store.vala

```diff
@@ -1,307 +1,658 @@
-    public E.SourceRegistry registry { get; private set; }
+    public signal void error_received (GLib.Error e);
 
-    /* The first day of the month */
-    public DateTime month_start { get; set; }
+    /* Notifies when sources are added, changed, or removed */
+    public signal void source_connecting (E.Source source, GLib.Cancellable cancellable);
+    public signal void source_added (E.Source source);
+    public signal void source_changed (E.Source source);
+    public signal void source_removed (E.Source source);
 
-    /* The number of weeks to show in this model */
-    public int num_weeks { get; private set; default = 6; }
-
-    /* The start of week, ie. Monday=1 or Sunday=7 */
-    public GLib.DateWeekday week_starts_on { get; set; default = GLib.DateWeekday.MONDAY; }
-
-    /* The event that is currently dragged */
-    public ECal.Component drag_component {get; set;}
-
-    /* Notifies when events are added, updated, or removed */
-    public signal void events_added (E.Source source, Gee.Collection<ECal.Component> events);
-    public signal void events_updated (E.Source source, Gee.Collection<ECal.Component> events);
-    public signal void events_removed (E.Source source, Gee.Collection<ECal.Component> events);
+    /* Notifies when components are added, modified, or removed */
+    public signal void components_added (Gee.Collection<ECal.Component> components, E.Source source, Gee.Collection<ECal.ClientView> views);
+    public signal void components_modified (Gee.Collection<ECal.Component> components, E.Source source, Gee.Collection<ECal.ClientView> views);
+    public signal void components_removed (Gee.Collection<ECal.Component> components, E.Source source, Gee.Collection<ECal.ClientView> views);
 
-    public signal void connecting (E.Source source, Cancellable cancellable);
-    public signal void connected (E.Source source);
-    public signal void error_received (string error);
+    public ECal.ClientSourceType source_type { get; construct; }
+    private E.SourceRegistry registry { get; private set; }
+    private HashTable<string, ECal.Client> source_client;
+    private HashTable<string, Gee.ArrayList<ECal.ClientView>> source_views;
 
-    /* The month_start, num_weeks, or week_starts_on have been changed */
-    public signal void parameters_changed ();
+    private HashTable<ECal.ClientView, Gee.Collection<ECal.Component>> components_add_transaction;
+    private Gee.Collection<E.Source> sources_add_transaction;
+    private Gee.Collection<E.Source> sources_connect_transaction;
 
-    HashTable<string, ECal.Client> source_client;
-    HashTable<string, ECal.ClientView> source_view;
-    HashTable<E.Source, Gee.TreeMultiMap<string, ECal.Component>> source_events;
+    internal HashTable<string, Gee.TreeMultiMap<string, ECal.Component>> source_components;
 
-    public GLib.Queue<E.Source> calendar_trash;
+    private GLib.Queue<E.Source> sources_trashed;
```

Comment:
Minor style: this is a property, not a function, so it should be a noun. This would also be consistent with `calendar_trash`.

```suggestion
    private GLib.Queue<E.Source> sources_trash;
```

Reply:
Fixed!